### PR TITLE
Remove optimism feature

### PR DESCRIPTION
**Pull request by Parikalp-Bhardwaj**
closes ##11534

## Description
- copy all the tests that have an `optimism` feature gate inside and move them into the 'optimism-evm' crate
- Removed `optimism` feature gate from 'optimism-evm' crate .

### CODE REVIEW

**Review by mattsse — COMMENTED** [review]
this is currently not possible until we have a proper abstraction over receipts/blocks etc..

**Comment by mattsse** [comment]
what we can do instead is move those to the optimism crate and feature gate your changes to not(optimism)

**Review by emhane — CHANGES_REQUESTED** [review]
this doesn't address the issue, try reading it again :) it says to copy the tests to op crate also

**Comment by Parikalp-Bhardwaj** [comment]
Hii @emhane, I have to copy all the tests that have an `optimism` feature gate inside from `reth/crates/evm/execution-types/src/execution_outcome.rs` and move them into the 'optimism-evm' crate at `reth/crates/optimism/evm/src/execution_outcome.rs`?

For the copies in the 'optimism-evm' crate, i need to remove the `optimism` feature gate

**Comment by emhane** [comment]
> Hii @emhane, I have to copy all the tests that have an `optimism` feature gate inside from `reth/crates/evm/execution-types/src/execution_outcome.rs` and move them into the 'optimism-evm' crate at `reth/crates/optimism/evm/src/execution_outcome.rs`?
> 
yes, but the tests can go in the already existing tests module in _reth/crates/optimism/evm/src/execute.rs_

> For the copies in the 'optimism-evm' crate, i need to remove the `optimism` feature gate

correct!

File: crates/optimism/evm/src/execute.rs [hunk 1]
```diff
@@ -158,12 +158,12 @@ where
                     transaction_gas_limit: transaction.gas_limit(),
                     block_available_gas,
                 }
-                .into())
+                .into());
```
Comment (emhane) [inline crates/optimism/evm/src/execute.rs:161]:
```suggestion
                .into())
```
pls revert all these unrelated formatting changes. don't run `cargo fmt` for reth. use the nightly toolchain `cargo +nightly fmt`.

**Review by emhane — CHANGES_REQUESTED** [review]

File: crates/optimism/evm/src/execute.rs (hunk 1 above)
Comment (Parikalp-Bhardwaj) [reply crates/optimism/evm/src/execute.rs:161]: @emhane, I have attempted to revert the formatting changes and used `cargo +nightly fmt` as you suggested, but I'm still facing issues. Could you please provide further guidance on how to resolve this?

File: crates/optimism/evm/src/execute.rs [hunk 2]
```diff
@@ -158,12 +158,12 @@ where
...
             }
 
             // An optimism block should never contain blob transactions.
             if matches!(transaction.tx_type(), TxType::Eip4844) {
-                return Err(OptimismBlockExecutionError::BlobTransactionRejected.into())
+                return Err(OptimismBlockExecutionError::BlobTransactionRejected.into());
```
Comment (emhane) [inline crates/optimism/evm/src/execute.rs:166]:
```suggestion
                return Err(OptimismBlockExecutionError::BlobTransactionRejected.into())
```

**Review by emhane — CHANGES_REQUESTED** [review]

**Review by emhane — APPROVED** [review]

**Comment by emhane** [comment]
seems this has already been solved
